src/tools/ResultsComparer/Program.cs

```diff
@@ -70,6 +72,28 @@ private static IEnumerable<(string id, Benchmark baseResult, Benchmark diffResul
             }
         }
 
+        private static void PrintSummary((string id, Benchmark baseResult, Benchmark diffResult, EquivalenceTestConclusion conclusion)[] notSame)
+        {
+            var better = notSame.Where(result => result.conclusion == EquivalenceTestConclusion.Faster);
+            var worse = notSame.Where(result => result.conclusion == EquivalenceTestConclusion.Slower);
+            var betterCount = better.Count();
+            var worseCount = worse.Count();
+
+            // If the baseline doesn't have the same set of tests, you wind up with Infinity in the list of diffs.
+            // Exclude them for purposes of geomean.
+            better = better.Where(x => GetRatio(x) != double.PositiveInfinity);
+            worse = worse.Where(x => GetRatio(x) != double.PositiveInfinity);
+
+            var betterGeoMean = Math.Pow(better.Skip(1).Aggregate(GetRatio(better.First()), (x, y) => x * GetRatio(y)), (1.0 / better.Count()));
```

Comment:
When computing the geometric mean you should use the relationship with logarithms to minimize the chances of overflow. More here: https://en.wikipedia.org/wiki/Geometric_mean#Relationship_with_logarithms

Reply:
To generalize the computation, you could also take into account zero values in different metrics (for example allocated memory in current thread metric for Span benchmarks).

Reply:
I'm guessing resolved as "Won't fix"?

Reply:
Correct.

Reply:
Apparently typing a comment and clicking resolve is the wrong flow. I typed up a reasonable reason. 🙂 I’ll see if it’s still on a text box at work or re type it once I get to a real keyboard.

Reply:
@jorive 
re: overflow, sure. good suggestion.

re: generalizing: the tool only cares about time/op right now, making it more generic would be a lot more work. I think that's a good note for the work being done on new tools. This is more of a stopgap to scratch the itch Andy and I both had.